foursight_core/react/api/react_ui.py

```diff
@@ -0,0 +1,103 @@
+import os
+import io
+
+REACT_BASE_DIR = os.path.join(os.path.dirname(__file__), "../../react/ui")
+REACT_DEFAULT_FILE = "index.html"
+REACT_WHITELISTED_FILE_PATH_SUFFIXES = [
+    "/index.html",
+    "/main.js",
+    "/main.css",
+    "/manifest.json",
+    "/asset-manifest.json",
+    ".jpeg",
+    ".jpg",
+    ".png",
+    ".ico"
+]
+
+
+class ReactUi:
+
+    def __init__(self, react_api):
+        self.react_api = react_api
+
+    class Cache:
+        static_files = {}
+
+    def serve_static_file(self, env: str, **kwargs):
+
+        # TODO: commentary on this.
+        env = env.replace("{environ}", env)
+
+        if env == "static":
+            # If the env is 'static' then we take this to mean the 'static'
+            # subdirectory; this is the directory where the static (js, css, etc)
+            # React files live. Note that this means a real 'environ' may not be 'static'.
+            file = os.path.join(REACT_BASE_DIR, "static")
+        else:
+            file = REACT_BASE_DIR
+        args = kwargs.values()
+        if not args:
+            # TODO: png (et.al.) not downloading right!
+            # Running chalice local it works though.
+            # Actually it also works in cgap-supertest:
+            # https://810xasmho0.execute-api.us-east-1.amazonaws.com/api/react/logo192.png
+            # But not in 4dn/foursight-development:
+            # https://cm3dqx36s7.execute-api.us-east-1.amazonaws.com/api/react/logo192.png
```

Comment:
Probably the thing to do here is look for any differences in the libraries across the deployments, if possible.